xcode/Subconscious/Shared/Components/AddressBook/AddressBookModel.swift

```diff
@@ -187,9 +200,17 @@ struct AddressBookModel: ModelProtocol {
                 return Update(state: state)
             }
             
+            guard !state.follows.contains(where: { f in f.did == did }) else {
+                return update(
+                    state: state,
+                    action: .failFollow(error: "Already following user"),
+                    environment: environment
+                )
+            }
+            
             let fx: Fx<AddressBookAction> =
                 environment.data.addressBook
-                .setPetnameAsync(did: did, petname: petname)
+                .followUserAsync(did: did, petname: petname)
```

Comment:
Higher-level concept: `follow` is user-level and `set` is an implementation detail